packages/react/src/components/cookieConsent/consentGroups/ConsentGroups.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { useContext } from 'react';
+import { VisuallyHidden } from '@react-aria/visually-hidden';
+
+import {
+  CookieConsentContext,
+  RequiredOrOptionalConsentGroups,
+  useCookieConsentActions,
+} from '../CookieConsentContext';
+import styles from '../CookieConsent.module.scss';
+import ConsentGroup from '../consentGroup/ConsentGroup';
+import { Checkbox } from '../../checkbox/Checkbox';
+
+function ConsentGroups(props: {
+  consentGroups?: RequiredOrOptionalConsentGroups;
+  isRequired?: boolean;
+}): React.ReactElement {
+  const { consentGroups, isRequired } = props;
+  const cookieConsentContext = useContext(CookieConsentContext);
+  const triggerAction = useCookieConsentActions();
+  if (!consentGroups) {
+    return null;
+  }
+  const selectPercentage = cookieConsentContext.countApprovedOptional();
+  const allApproved = isRequired || selectPercentage === 1;
+  const { title, text, groupList, checkboxAriaDescription } = consentGroups;
+  const groupId = isRequired ? 'required' : 'optional';
```

Comment:
I would move this line 26 (const groupId = ...) to line 37 to improve readability, as the groupId is only used in line 38. It's easier to read when things that are related are more close to each other.